modules/cmsadmin/blocks/VideoBlock.php

```diff
@@ -20,57 +20,76 @@ public function config()
     {
         return [
             'vars' => [
-                ['var' => 'url', 'label' => 'Video URL', 'type' => 'zaa-text'],
+                ['var' => 'url', 'label' => 'Video URL (Vimeo oder Youtube)', 'type' => 'zaa-text'],
             ],
             'cfgs' => [
-                ['var' => 'controls', 'label' => 'Controls ausblenden?', 'initvalue' => '1', 'type' => 'zaa-select', 'options' => [
-                        ['value' => '0', 'label' => 'Nein'],
-                        ['value' => '1', 'label' => 'Ja'],
-                    ],
+                [
+                    'var' => 'controls',
+                    'label' => 'Controls ausblenden? (Wird nicht von allen Playern unterstützt)',
+                    'type' => 'zaa-checkbox',
                 ],
             ],
         ];
     }
 
-    public function constructUrl()
+    public function constructYoutubeUrl($scheme, $host, $path, $query)
     {
-        $scheme = parse_url($this->getVarValue('url'), PHP_URL_SCHEME);
-        $host = parse_url($this->getVarValue('url'), PHP_URL_HOST);
-        $path = parse_url($this->getVarValue('url'), PHP_URL_PATH);
-        $query = parse_url($this->getVarValue('url'), PHP_URL_QUERY);
-
-        // youtube embed url constructing
-
         if (($pos = strpos($query, 'v=')) !== false) {
             $videoId = substr($query, $pos + 2);
 
-            $url = $scheme.'://'.$host.'/embed/'.$videoId;
+            $url = $scheme . '://' . $host . '/embed/' . $videoId;
 
-            if ($this->getCfgValue('controls') == 1) {
+            if ($this->getCfgValue('controls')) {
                 $url .= '?rel=0&amp;controls=0&amp;showinfo=0';
+            } else {
+                $url .= '?' . $this->getCfgValue('controls');
             }
         } else {
-            // not supported yet
             $url = '';
         }
+        return $url;
+    }
 
+    public function constructVimeoUrl($scheme, $host, $path, $query)
+    {
+        $path = trim($path, '/');
+        $url = '';
+        if (is_numeric($path)) {
+            $url = 'https://player.vimeo.com/video/' . $path . '?color=0c88dd&title=0&byline=0&portrait=0&badge=0';
+        }
         return $url;
     }
 
+    public function constructUrl()
+    {
+        $scheme = parse_url($this->getVarValue('url'), PHP_URL_SCHEME);
+        $host = parse_url($this->getVarValue('url'), PHP_URL_HOST);
+        $path = parse_url($this->getVarValue('url'), PHP_URL_PATH);
+        $query = parse_url($this->getVarValue('url'), PHP_URL_QUERY);
+
+        if ($host == 'www.youtube.com') {
+            return $this->constructYoutubeUrl($scheme, $host, $path, $query);
+        } elseif (($host == 'www.vimeo.com') || (($host == 'vimeo.com'))) {
+            return $this->constructVimeoUrl($scheme, $host, $path, $query);
+        } else {
+            return '';
```

Comment:
the last else is not needed ... and you should return null or false and not an empty string. thats what i mean, sorry for the bad explanation.